### PR TITLE
Fix ineffectual assignments

### LINKED ISSUE #14943: Ineffectual Go variable assignments
This repo has ineffectual assignments in go code: https://goreportcard.com/report/github.com/kubernetes/test-infra#ineffassign

We should resolve these errors by identifying and correcting to the intended assignment or, at minimum, removing the ineffectual assignment.
/help wanted

### PR DESCRIPTION
**Pull request by ibrasho**
Fix ineffectual assignments reported in https://goreportcard.com/report/github.com/kubernetes/test-infra#ineffassign

Fixes #14943

### CODE REVIEW

File: def/configmap/main.go [hunk 1]
```diff
@@ -56,17 +56,16 @@ func (mkv *multiKeyValue) String() string {
...
+	}
+
 	p := strings.SplitN(v, "=", 2)
 	if len(p) != 2 {
 		return fmt.Errorf("%s does not match label=value", v)
 	}
-	if mkv == nil {
-		mkv = &multiKeyValue{
-			p[0]: p[1],
-		}
```
Comment (ibrasho) [inline def/configmap/main.go]: This doesn't work. You can't assign a value to a pointer receiver and expect the original value pointed at to be updated.

File: prow/flagutil/github.go [hunk 2]
```diff
@@ -63,10 +63,10 @@ func (o *GitHubOptions) addFlags(wantDefaultGitHubTokenPath bool, fs *flag.FlagS
 // Validate validates GitHub options.
 func (o *GitHubOptions) Validate(dryRun bool) error {
 	for _, uri := range o.endpoint.Strings() {
-		if uri == "" {
-			uri = github.DefaultAPIEndpoint
```
Comment (ibrasho) [inline prow/flagutil/github.go]: `uri` is not used after this assignment.

File: prow/repoowners/repoowners.go [hunk 3]
```diff
@@ -686,8 +686,7 @@ func (o *RepoOwners) entriesForFile(path string, people map[string]map[*regexp.R
 	d := path
 	if !o.enableMDYAML || !strings.HasSuffix(path, ".md") {
 		// if path is a directory, this will remove the leaf directory, and returns "." for topmost dir
-		d = filepath.Dir(d)
-		d = canonicalize(path)
+		d = canonicalize(filepath.Dir(path))
```
Comment (ibrasho) [inline prow/repoowners/repoowners.go]: I'm not sure if this the correct value yet.

File: prow/pluginhelp/hook/hook.go [hunk 4]
```diff
@@ -361,13 +361,14 @@ func (ha *HelpAgent) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "405 Method not allowed", http.StatusMethodNotAllowed)
 		return
 	}
-	b, err := ioutil.ReadAll(r.Body)
-	if err != nil {
+
+	if _, err := ioutil.ReadAll(r.Body); err != nil {
```
Comment (cjwagner) [inline prow/pluginhelp/hook/hook.go]: We shouldn't be reading the request body at all here (it is a GET request). Please remove this entirely.

File: prow/repoowners/repoowners_test.go [hunk 5]
```diff
@@ -1264,11 +1264,10 @@ reviewers:
...
 			t.Errorf("result '%s' is differ from expected: '%s'", s, test.expected)
 		}
-		simple, err := LoadSimpleConfig(b)
-		if err != nil {
+
+		if simple, err := LoadSimpleConfig(b); err != nil {
 			t.Errorf("unexpected error when load simple config: %v", err)
-		}
-		if !reflect.DeepEqual(simple, test.given) {
+		} else if !reflect.DeepEqual(simple, test.given) {
```
Comment (cjwagner) [inline prow/repoowners/repoowners_test.go]: What is this change fixing? AFAICT both `simple` and `err` were already being assigned properly.

Comment (ibrasho) [reply prow/repoowners/repoowners_test.go]: I don't really remember why I made this change, but as you said, it's not needed.

File: prow/pluginhelp/hook/hook.go (hunk 4 above)
Comment (ibrasho) [reply prow/pluginhelp/hook/hook.go]: Done.

**Pull request merged**